src/common/hist_util.cu

```diff
@@ -324,9 +316,27 @@ struct GPUSketcher {
 
       // add cuts into sketches
       thrust::copy(cuts_d_.begin(), cuts_d_.end(), cuts_h_.begin());
+#pragma omp parallel for schedule(static)
       for (int icol = 0; icol < num_cols_; ++icol) {
-        summaries_[icol].MakeFromSorted(&cuts_h_[n_cuts_ * icol], n_cuts_cur_[icol]);
-        sketches_[icol].PushSummary(summaries_[icol]);
+        WXQSketch::SummaryContainer summary;
+        summary.Reserve(n_cuts_);
+        summary.MakeFromSorted(&cuts_h_[n_cuts_ * icol], n_cuts_cur_[icol]);
+
+        // Spin in user land until we we can grab this lock
+        struct ColLocker {
```

Comment:
Please use `std::mutex`.

Reply:
done. i did this to avoid context switching when contention is greater and the work done in the critical section is smaller. i have changed this to keep it simple for now.